worker-core/src/test/java/com/hpe/caf/worker/core/StreamingWorkerWrapperTest.java

```diff
@@ -212,6 +212,40 @@ public void testAbandon()
         Assert.assertEquals(queueMsgId, callback.getQueueMsgId());
     }
 
+    @Test
+    public void testPoisoned()
+        throws InvalidTaskException, TaskRejectedException, InterruptedException, InvalidNameException, CodecException
+    {
+        Codec codec = new JsonCodec();
+        WorkerFactory happyWorkerFactory = mock(WorkerFactory.class);
+        Worker happyWorker = Mockito.spy(getWorker(new TestWorkerTask(), codec));
+        when(happyWorkerFactory.getWorker(Mockito.any())).thenReturn(happyWorker);
+        when(happyWorker.doWork()).thenAnswer(invocationOnMock -> {
+            throw new TaskFailedException("whoops");
+        });
+        MessagePriorityManager priorityManager = mock(MessagePriorityManager.class);
+        when(priorityManager.getResponsePriority(Mockito.any())).thenReturn(PRIORITY);
+        String queueMsgId = "exception";
+        CountDownLatch latch = new CountDownLatch(1);
+        TestCallback callback = new TestCallback(latch);
+        TaskMessage m = new TaskMessage();
+        ServicePath path = new ServicePath(SERVICE_NAME);
+        Map<String, byte[]> contextMap = new HashMap<>();
+        contextMap.put(path.toString(), SUCCESS_BYTES);
+        m.setTaskId(TASK_ID);
+        m.setContext(contextMap);
+        m.setTaskData("Test data".getBytes(StandardCharsets.UTF_8));
```

Comment:
The taskdata might not be a byte representation of a string (which would always contain printable characters)
Test this with more challenging binary data (and let me know if there is a problem or not)